### DISCUSSION TITLE
How to change the password in auth0

### DISCUSSION THREAD

**Discussion by KetulCodiste** [Web3Auth/Web3Auth#574] [Help]
I am using the web3Auth + Auth0
I have successfully implemented the flow of login in my application. Now if my user want to change the password, then I want to have an option of changing password in my website's setting page. how would my user will be able to  change the password?

**Comment by rinzler99** [comment]
If you are using username & password(db connection flow) in auth0, You can have a button in your application which will redirect to auth0 reset password link.

Pls refer below link for more info:
https://auth0.com/docs/authenticate/database-connections/password-change#trigger-an-interactive-password-reset-flow

**Reply by KetulCodiste** [reply]
My expectation:
I have a "Change Password" button in my website. When the user clicks, the Auth0 popup should open asking for the email to change the password. Then user will get the link in the email and from there, he will be able to reset the password.

What my current flow is:
When user clicks the "Change Password" button, the login screen is popping up having the email and password and below this there is an option "Forgot Password" which the user can click and then their is the field asking  for the email to change the password.

You can see that there is an extra step for the user of clicking the "Forget Password" button to change the password

U can see that even after clicking the "Change Password" button, user needs to click one more button to change the password, which can make them confuse.

So how should I fix this issue?

**Reply by rinzler99** [reply]
To avoid the extra step you can refer the above mentioned auth0 doc link [url](https://auth0.com/docs/authenticate/database-connections/password-change#trigger-an-interactive-password-reset-flow):
1. You can send the email for password reset from autho using below POST api call.
<img width="810" alt="image" src="https://user-images.githubusercontent.com/30899350/189466681-e81b896a-b03a-46af-9c7e-518d730c79dc.png">

2. For this you need to create a ui where user will enter the email and pass that email to above POST api call. After the post request you will get the below email with link to reset password.
<img width="644" alt="image" src="https://user-images.githubusercontent.com/30899350/189466782-4936f56d-b7ae-405d-98e3-297af82a520b.png">
